File: tests/java/tests-java-gradle-junit5-allure3-selenium/src/test/java/tests/e2e/HeaderTests.java
package tests.e2e;

import tests.TestBase;
import annotations.Layer;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

@Layer("e2e")
@Epic("Header")
@Feature("Lang and theme")
@Severity(SeverityLevel.NORMAL)
@DisplayName("Header")
class HeaderTests extends TestBase {

    @Test
    @Tag("e2e")
    @DisplayName("Login page stays English by default")
    void loginPageStaysEnglishByDefault() {
        loginPage.openPage()
                .shouldHaveFormTitle("Login Form")
                .header.shouldHaveLangLabel("EN")
                .shouldHaveHtmlLang("en");
    }

    @Test
    @Tag("e2e")
    @DisplayName("Theme toggle persists light theme after reload")
    void themeTogglePersistsLightThemeAfterReload() {
        loginPage.openPage()
                .shouldHaveFormTitle("Login Form")
                .header.shouldHaveTheme("dark")
                .clickThemeToggle()
                .shouldHaveTheme("light");
        loginPage.reloadPage()
                .header.shouldHaveTheme("light");
    }

    @Test
    @Tag("e2e")
    @DisplayName("Lang toggle switches login copy to Russian and back")
    void langToggleSwitchesLoginCopyToRussianAndBack() {
        loginPage.openPage()
                .shouldHaveFormTitle("Login Form")
                .header.clickLangToggle()
                .shouldHaveLangLabel("RU")
                .shouldHaveHtmlLang("ru");
        loginPage.shouldHaveFormTitle("Форма входа")
                .reloadPage()
                .header.shouldHaveLangLabel("RU")
                .shouldHaveHtmlLang("ru");
        loginPage.shouldHaveFormTitle("Форма входа")
                .header.clickLangToggle()
                .shouldHaveLangLabel("EN")
                .shouldHaveHtmlLang("en");
        loginPage.shouldHaveFormTitle("Login Form");
    }
}

File: tests/rust/tests-rust-testing-reqwest-selenium/tests/ui/header_tests.rs
use allure_cargotest::allure_test;
use tests_rust_testing_reqwest_selenium as tests;

fn layer() {
    tests::layer_ui("Header", "Header", "Lang and theme", "normal");
}

#[allure_test(name = "Login page stays English by default")]
#[tokio::test]
async fn login_page_stays_english_by_default() {
    layer();
    tests::with_browser(|| async {
        tests::LoginPage::default()
            .open_page()
            .await
            .should_have_form_title("Login Form")
            .await
            .header
            .should_have_lang_label("EN")
            .await
            .should_have_html_lang("en")
            .await;
    })
    .await;
}

#[allure_test(name = "Theme toggle persists light theme after reload")]
#[tokio::test]
async fn theme_toggle_persists_light_theme_after_reload() {
    layer();
    tests::with_browser(|| async {
        tests::LoginPage::default()
            .open_page()
            .await
            .should_have_form_title("Login Form")
            .await
            .header
            .should_have_theme("dark")
            .await
            .click_theme_toggle()
            .await
            .should_have_theme("light")
            .await;
        tests::LoginPage::default()
            .reload_page()
            .await
            .header
            .should_have_theme("light")
            .await;
    })
    .await;
}

#[allure_test(name = "Lang toggle switches login copy to Russian and back")]
#[tokio::test]
async fn lang_toggle_switches_login_copy_to_russian_and_back() {
    layer();
    tests::with_browser(|| async {
        tests::LoginPage::default()
            .open_page()
            .await
            .should_have_form_title("Login Form")
            .await
            .header
            .click_lang_toggle()
            .await
            .should_have_lang_label("RU")
            .await
            .should_have_html_lang("ru")
            .await;
        tests::LoginPage::default()
            .should_have_form_title("Форма входа")
            .await
            .reload_page()
            .await
            .header
            .should_have_lang_label("RU")
            .await
            .should_have_html_lang("ru")
            .await;
        tests::LoginPage::default()
            .should_have_form_title("Форма входа")
            .await
            .header
            .click_lang_toggle()
            .await
            .should_have_lang_label("EN")
            .await
            .should_have_html_lang("en")
            .await;
        tests::LoginPage::default()
            .should_have_form_title("Login Form")
            .await;
    })
    .await;
}
